ckanext/datadotworld/api.py

```diff
@@ -120,6 +122,17 @@ def _prepare_resource_url(res):
 
     return prepared_data
 
+def _delay_request():
```

Comment:
Correct me if I'm wrong, but this doesn't accomplish rate limiting it just postpones request bursts by 1 second. Instead, proper rate limiting and retries on HTTP 429 was what I had in mind:
- Rate limiting: https://pypi.python.org/pypi/ratelimit
- Retries: https://stackoverflow.com/questions/23267409/how-to-implement-retry-mechanism-into-python-requests-library

Reply:
@rflprr every dataset/resource update/create action is added to a queue and processed separately from other items. Therefore, we only could add a delay between jobs processing in a queue. As we set a delay to 1sec by default, this means that only 1 dataset (which is usually only 1 API call) will be processed every second. And therefore there will be no request bursts.

Reply:
@rflprr but I noticed an issue. Currently, on 429 response Dataset's status stays as Pending, but the job is not re-added to the queue for a re-try. We are adding a fix now. It should be ready tomorrow, so please postpone merging the PR. I will update it soon.